manifests/postgres.pp

```diff
@@ -173,7 +173,7 @@
   # define user used by Barman to connect into PostgreSQL database(s)
   postgresql::server::role { $barman_dbuser:
     login         => true,
-    password_hash => postgresql_password($barman_dbuser, $real_password),
+    password_hash => postgresql_password($barman_dbuser, inline_template('<%= @real_password.to_s %>')),
```

Comment:
This was the best suggestion that I got from the Puppet Users group.  It gets the desired result and puppet-lint doesn't throw any errors.  Do you have another idea to help get this module be compatible with puppet > 4.x?